code/datums/components/mood.dm

```diff
@@ -59,36 +103,66 @@
 		owner.hud_used.mood.icon_state = "mood[mood_level]"
 
 /datum/component/mood/process() //Called on SSmood process
-	switch(mood)
-		if(-INFINITY to MOOD_LEVEL_SAD4)
+	switch(sanity)
+		if(SANITY_INSANE to SANITY_CRAZY)
 			owner.overlay_fullscreen("depression", /obj/screen/fullscreen/depression, 3)
-		if(MOOD_LEVEL_SAD4 to MOOD_LEVEL_SAD3)
+		if(SANITY_INSANE to SANITY_UNSTABLE)
 			owner.overlay_fullscreen("depression", /obj/screen/fullscreen/depression, 2)
-		if(MOOD_LEVEL_SAD3 to MOOD_LEVEL_SAD2)
+		if(SANITY_UNSTABLE to SANITY_DISTURBED)
 			owner.overlay_fullscreen("depression", /obj/screen/fullscreen/depression, 1)
-		if(MOOD_LEVEL_SAD2 to INFINITY)
+		if(SANITY_DISTURBED to SANITY_GREAT)
 			owner.clear_fullscreen("depression")
 
+	switch(mood_level)
+		if(1)
+			DecreaseSanity(0.4)
+		if(2)
+			DecreaseSanity(0.25)
+		if(3)
+			DecreaseSanity(0.15)
+		if(4)
+			DecreaseSanity(0.05)
+		if(5)
+			IncreaseSanity(0.1)
+		if(6)
+			IncreaseSanity(0.15)
+		if(7)
+			IncreaseSanity(0.20)
+		if(8)
+			IncreaseSanity(0.25, 125)
+		if(9)
+			IncreaseSanity(0.4, 125)
+
 	if(owner.has_trait(TRAIT_DEPRESSION))
-		if(prob(0.1))
+		if(prob(0.05))
 			add_event("depression", /datum/mood_event/depression)
 			clear_event("jolly")
 	if(owner.has_trait(TRAIT_JOLLY))
-		if(prob(0.1))
+		if(prob(0.05))
 			add_event("jolly", /datum/mood_event/jolly)
 			clear_event("depression")
 
+	var/area/A = get_area(owner)
+	if(A)
+		update_beauty(A)
+
+/datum/component/mood/proc/DecreaseSanity(amount)
+	sanity = max(0, sanity - amount)
+
+/datum/component/mood/proc/IncreaseSanity(amount, limit = 99)
+	if(sanity > limit)
+		DecreaseSanity(-0.5) //Removes some sanity to go back to our current limit.
```

Comment:
why, you're doing a min on the very next line

Reply:
If you are high sanity, for example, 125, and you suddenly become sad, I don't want your sanity jumping from 125 to 100 due to the min(). This instead makes your sanity go back down to the limit at a pace of 0.5 per tick.

(I'm not sure if there's a better way of handling that, though)

Reply:
It doesn't though, it decreases your sanity by 0.5 and then the very next line sets it to the minimum.

it's not an if else and you dont' have an early return so both lines are running?

Reply:
oh yeah disregard me im retarded I should be returning, thanks.